Aromatics updates 2 (out of 4): Thermo Groups

zjburas opened the pull request:
New thermo groups, mostly ring() and polycyclic() corrections, for species that were found to be important in modeling 1,3-CPD and Natural Gas pyrolysis to form aromatics. New groups mostly based on either M06 or CBS-QB3 calculations. Also added a few globally forbidden species that were problematic due to resonance and unlikely to form anyway.

For the ring() corrections, tree was re-organized slightly, so that even if a strained cyclic species doesn't have an exact match (four consecutive double bonds in a 6-membered ring, for example), it will at least use a cyclic correction that provides a reasonable lower bound on the strain (three consecutive double bonds instead, for example), rather than simply falling up to the most general node.

@KEHANG might be the best reviewer for this PR.

Builds off of #205

KEHANG commented:
Overall it looks good to me. Some small details we might need to discuss this coming week in-person.

zjburas commented:
In response to @KEHANG 's comments:

1. Some repetitive commits squashed
2. Fixed a typo in one of the commits
3. Deleted some unnecessary commits creating small thermo libraries

As a reminder, this PR should be merged along with @KEHANG 's new bicyclic thermo estimator algorithm (no PR yet) to get the full benefit of these changes.

KEHANG commented:
The RMG-Py PR will come soon

KEHANG commented:
Since this PR is mostly for adding more thermo data to RMG-database, we have to estimate how much improvement it brings. At least, it should not make the performance we've already achieved in current master branch. 

In order to do that, I did a prediction performance comparison between two versions: RMG-Py master with RMG-database master and RMG-Py bicyclic_formula branch with RMG-database Aromatics_update_2 branch. The data tables are prepared by myself since last year (the number is increasing ever since). Results are shown below.

We expected this PR to help predict highly unsaturated polycyclics (`zb_bicyclics_195_table`) without affecting highly saturated polycyclics (all the other tables) predictions. And it is exactly what we saw from the test results. So I agree to merge this PR into master.

 thermo estimation error (kcal/mol) | master-rmgpy & master-db | bicyclic_formula-rmgpy & Aromatics_updates_2-db 
----- | ----- | -----
small_cyclic_table | 3.39 | 3.42 
large_linear_polycyclic_table | 9.10 | 9.05 
large_fused_polycyclic_table | 9.56 | 9.54
small_O_only_polycyclic_table | 6.93 | 6.95
large_linear_O_only_polycyclic_table | 10.31 | 10.31
large_fused_O_only_polycyclic_table | 5.33 | 5.30
zb_bicyclics_195_table | 62.89 | 9.59

Side note: The thermo validation functionality should be automated and be part of RMG-tests, which is coming soon.
